pittapi/dining.py

```diff
@@ -1,98 +1,146 @@
-#    try:
-#        return get_locations()[location]
-#    except:
-#        raise ValueError('The dining location is invalid')
-
-# def get_location_menu(location=None, date=None):
-# location can only be market, market's subordinates, and cathedral cafe
-# if location is none, return all menus, and date will be ignored
-# date has to be a day of the week, or if empty will return menus for all days of the week
-#
-# https://www.pc.pitt.edu/dining/menus/flyingStar.php
-# https://www.pc.pitt.edu/dining/menus/bellaTrattoria.php
-# https://www.pc.pitt.edu/dining/menus/basicKneads.php
-# https://www.pc.pitt.edu/dining/menus/basicKneads.php
-# https://www.pc.pitt.edu/dining/menus/magellans.php
-# https://www.pc.pitt.edu/dining/locations/cathedralCafe.php
-#    return []
+                start_time = datetime.time(
+                    j["hours"][0]["start_hour"], j["hours"][0]["start_minutes"])
+                end_time = datetime.time(
+                    j["hours"][0]["end_hour"], j["hours"][0]["end_minutes"])
+
+            week_info.append([datetime.datetime(int(date[0:4]), int(
+                date[5:7]), int(date[8:])), j["status"], start_time, end_time])
+
+        hours[i["name"]] = week_info
+
+    return hours
+
+
+def get_menu(location, date=datetime.datetime.today()):
+    """
+    location will be string of location name
+    date must be in YYYY-MM-DD format
+
+    Returns:
+        None if location is closed
+        
+
+    Raises Exception:
+        If req status is unsuccessful
+        Usually occurs when no menu is found
+    """
+    menu = {}
+    url = "https://api.dineoncampus.com/v1/location/"
+
+    loc_id = get_all_locations()[location]
+
+    url += loc_id
+    url += "/periods?platform=0&date="
+
+    # could probably restrict arguments
+    if(type(date) == datetime.datetime):
+        date = date.isoformat()
+        url += date
+    else:
+        url += date
+
+    req = requests.get(url).json()
+
+    if req["status"] == "success":
+        if req["closed"] == True:
+            return None
+    else:
+        raise Exception(req["msg:"])
+    
+    period_food = {} # is a dictionary containing the period (all day, breakfast, lunch, dinner) as value and the menu as a list 
+    for periods in req["periods"]:
+        period_id = periods["id"]
+        new_url = "https://api.dineoncampus.com/v1/location/" + loc_id + "/periods/" + period_id + "?platform=0&date=" + date
+
+        new_request = requests.get(new_url).json()
+
+        food = []
+        for i in new_request["menu"]["periods"]["categories"]:
+            for j in i["items"]:
+                food.append(j["name"])
+        
+        period_food[periods["name"]] = food
+    
+    menu[location] = period_food #returns a dictionary with the location as the key and a dictionary (see period_food) as the value
+
+    return menu
+
+
+
+
+
+
+# arguments don't really matter
+def get_all_locations(menus=True, address=False, buildings=True) -> Dict[str, str]:
```

Comment:
```suggestion
def get_all_locations(menus: bool = True, address: bool = False, buildings: bool = True) -> dict[str, str]:
```

1) Type hints
2) `Dict` -> `dict` (see previous comment)